src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { createContext, PropsWithChildren, useContext, useMemo, useState } from "react";
+import { Toast, ToastProps } from "./Toast";
+
+export type ToastContextProps = {
+  setNotice: React.Dispatch<React.SetStateAction<ToastProps | undefined>>;
+};
+
+export const ToastContext = createContext<ToastContextProps>({
+  setNotice: () => {
+    throw new Error("Missing ToastProvider");
+  },
+});
+
+export function ToastProvider(props: PropsWithChildren<{}>) {
+  const [notice, setNotice] = useState<ToastProps>();
+  const contextValue = useMemo(() => ({ setNotice }), []);
+  return (
+    <ToastContext.Provider value={contextValue}>
+      {props.children}
+      {notice && <Toast message={notice.message} type={notice.type} />}
```

Comment:
**thought:** I don't think this is how we want to deliver this `<Toast />` component. Instead, The `<Toast />` component should be something that the application can place anywhere. Right now we're forcing it to be underneath the rest of the page's content. I think we will probably want the user to be able to just drop in `<Toast />` and then that component is what would conditionally render based on `notice` being set.
I am thinking we might need to have the `ToastContextProps` include both `notice` and `setNotice`. Then we'd set the contextValue as: 
```
const contextValue = useMemo(() => ({ notice, setNotice }), [notice, setNotice]);
```

And in the `<Toast />` component, we'd remove the `ToastProps`, as we'll get everything from the hook now....
```
const { notice, setNotice } = useToast();
return <>{notice && (<div>...</div>)}</>;
```

Reply:
This suggestion unfortunately changes up a lot of this PR, and I'm sorry to not have noticed it/caught it until now. But I do think we need to change it so we can place the `Toast` component in the right place in Blueprint.

Reply:
no worries, I think that makes sense. I'll work on this tomorrow

Reply:
can we discuss this next week? I started implementing and am a little confused on how `useToast` is supposed to work..

Reply:
I think I figured it out!! took a lot of rubber ducking hah

Reply:
@bdow does the new implementation work?